### PR TITLE
Adds new tests to macro to get 100% coverage

### PR DESCRIPTION
**Pull request by umbernhard**
**Description**
Adds new tests to macro to get 100% coverage

**Testing**

New tests for corner cases in MACRO like recounts and worst-case scenarios. 

**Progress**
Ready

### CODE REVIEW

File: server/tests/audit_math/test_macro.py [hunk 1]
```diff
@@ -138,6 +138,48 @@ def test_max_error(contests, batches):
             )
 
 
+def test_get_sizes_extra_contests(contests, batches):
```
Comment (jonahkagan) [inline server/tests/audit_math/test_macro.py:141]: not sure i understand what this test is testing - can you explain?

File: server/tests/audit_math/test_macro.py [hunk 2]
```diff
@@ -208,6 +250,83 @@ def test_get_sample_sizes(contests, batches):
...
 
 
+def test_full_recount(contests, batches):
+    # Do a full recount:
+    sample = batches
+    expected_recount_size = {
+        "Contest A": 1,
+        "Contest B": 1,
+        "Contest C": 1,
+    }
```
Comment (jonahkagan) [inline server/tests/audit_math/test_macro.py:260]: why would a full recount only result in a sample size of one batch? i would expect the sample size to be the total number of batches

Comment (umbernhard) [reply server/tests/audit_math/test_macro.py:260]: This is default behavior for the sake of being conservative. In this test we're passing in our batches as the sample, so we've already done a full recount but we're asking for more ballots. I could change it to throw an error like I have done elsewhere.

File: server/tests/audit_math/test_macro.py (hunk 1 above)
Comment (umbernhard) [reply server/tests/audit_math/test_macro.py:141]: this is similar to what's going on in the `sampler` tests. There's a branch we need to test where the collected possible error is zero, so we achieve that by adding a contest that is only in some batches. If we compute the error for that contests in a batch that doesn't contain it, we get zero.

File: server/tests/audit_math/test_macro.py (hunk 2 above)
Comment (jonahkagan) [reply server/tests/audit_math/test_macro.py:260]: ah i get it. didn't see that we had already done the full recount. yes, i like the consistency of erroring here

File: server/tests/audit_math/test_macro.py [hunk 3]
```diff
@@ -259,15 +259,17 @@ def test_full_recount(contests, batches):
...
+        with pytest.raises(ValueError, match=r"All ballots have already been counted!"):
+            computed = macro.get_sample_sizes(
+                RISK_LIMIT, contests[contest], batches, sample
+            )
+
+            assert (
+                expected_recount_size[contest] == computed
+            ), "Third round sample expected {}, got {}".format(
+                expected_recount_size[contest], computed
+            )
```
Comment (jonahkagan) [inline server/tests/audit_math/test_macro.py:272]: i think this line is unreachable now, right?

**Review by jonahkagan — APPROVED** [review]
